tokman/app.py

```diff
@@ -0,0 +1,110 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import logging
+import os
+
+from datetime import datetime
+from pathlib import Path
+from flask import Flask, current_app
+from flask_sqlalchemy import SQLAlchemy
+from github import GithubIntegration
+
+try:
+    from flask_restx import Api, Resource
+except ModuleNotFoundError:
+    from flask_restplus import Api, Resource
+
+
+api = Api()
+db = SQLAlchemy()
+github_integration = None
+token_renew_at = None
+
+
+def create_app():
+    app = Flask(__name__)
+    app.config.from_envvar("TOKMAN_CONFIG")
+
+    log_level = os.getenv("LOG_LEVEL", "info")
+    log_level = getattr(logging, log_level.upper())
+    logging.basicConfig(level=log_level)
+
+    private_key = Path(app.config["GITHUB_APP_PRIVATE_KEY"]).read_text()
+    app_id = int(app.config["GITHUB_APP_ID"])
+    app.github_integration = GithubIntegration(app_id, private_key)
+
+    api.init_app(app)
+    db.init_app(app)
+
+    return app
+
+
+class Token(db.Model):
+    __tablename__ = "tokens"
+    id = db.Column(db.Integer, primary_key=True)
+    repo = db.Column(db.String, unique=True, nullable=False)
+    token = db.Column(db.String, nullable=True)
+    expires_at = db.Column(db.DateTime, nullable=True)
+
+    def is_expired(self):
+        token_renew_at = int(current_app.config.get("TOKEN_RENEW_AT", 60))
+        return (
+            self.expires_at is None
+            or self.token is None
```

Comment:
can those two happen independently?
- new token: `None, None`
- existing token: `expires_at, token`

Reply:
With the current implementation of how tokens are saved in the DB, no, it cannot happen independently.

But these values are stored in separate fields, so in theory it can happen (there is no strong guarantee that it wont, a future change could introduce this). As in the context of this method both not having an expiration date and not having a token can be interpreted as the token being expired, I think it makes sense to check them independently.